v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/options/ElasticsearchWriteOptions.java

```diff
@@ -73,4 +73,64 @@ public interface ElasticsearchWriteOptions extends PipelineOptions {
   Long getMaxRetryDuration();
 
   void setMaxRetryDuration(Long maxRetryDuration);
+
+  @Description("A property in the document being indexed whose value specifies which index to add the doc to (takes precendence over an index UDF)")
+  String getPropertyAsIndex();
+
+  void setPropertyAsIndex(String propertyAsIndex);
+
+  @Description("GCS path to JavaScript udf source for index extraction function")
+  String getJavaScriptIndexFnGCSPath();
+
+  void setJavaScriptIndexFnGCSPath(String javaScriptTextTransformGCSPath);
+
+  @Description("UDF JavaScript Function Name for index extraction")
+  String getJavaScriptIndexFnName();
+
+  void setJavaScriptIndexFnName(String javaScriptTextTransformFunctionName);
+
+  @Description("A property in the document being indexed whose value specifies which _id to use for the doc (takes precendence over an id UDF)")
+  String getPropertyAsId();
+
+  void setPropertyAsId(String propertyAsId);
+
+  @Description("GCS path to JavaScript udf source for id extraction function")
+  String getJavaScriptIdFnGCSPath();
+
+  void setJavaScriptIdFnGCSPath(String javaScriptTextTransformGCSPath);
+
+  @Description("UDF JavaScript Function Name for id extraction")
+  String getJavaScriptIdFnName();
+
+  void setJavaScriptIdFnName(String javaScriptTextTransformFunctionName);
+
+  @Description("GCS path to JavaScript udf source for type extraction function")
+  String getJavaScriptTypeFnGCSPath();
+
+  void setJavaScriptTypeFnGCSPath(String javaScriptTextTransformGCSPath);
+
+  @Description("UDF JavaScript Function Name for type extraction")
+  String getJavaScriptTypeFnName();
+
+  void setJavaScriptTypeFnName(String javaScriptTextTransformFunctionName);
+
+  @Description("GCS path to JavaScript udf source for deletion determination")
+  String getJavaScriptIsDeleteFnGCSPath();
+
+  void setJavaScriptIsDeleteFnGCSPath(String javaScriptTextTransformGCSPath);
+
+  @Description("UDF JavaScript Function Name for deletion determination")
+  String getJavaScriptIsDeleteFnName();
+
+  void setJavaScriptIsDeleteFnName(String javaScriptTextTransformFunctionName);
+
+  @Description("Whether to use partial updates (update type) with Elasticsearch requests")
+  Boolean getUsePartialUpdate();
+
+  void setUsePartialUpdate(Boolean usePartialUpdate);
+
+  @Description("Whether to use index (supports upsert) rather than create (errors on duplicate _id) in bulk requests")
+  Boolean getUseBulkIndexRatherThanCreate();
+
+  void setUseBulkIndexRatherThanCreate(Boolean useBulkIndexRatherThanCreate);
```

Comment:
nit: Maybe make it an Enum-type parameter called "bulkInsertMethod" or something like this, with CREATE and INDEX possible values, CREATE being the default?

Reply:
made the recommended change, look ok?